api/controllers/FileDownloadController.js

```diff
@@ -0,0 +1,66 @@
+const sails = require('sails');
+const request = require('request');
+const crypto = require('crypto');
+
+const FileDownloadController = {
+    downloadFileHandler(req, res) {
+        try {
+            const apiOptions = FileDownloadController._prepareAPIOptions(req);
+            FileDownloadController._streamFileToClient(apiOptions, res);
+        } catch (err) {
+            sails.log.error(err);
+            return res.serverError();
+        }
+    },
+
+    _prepareAPIOptions(req) {
+        if (req.params.apostilleRef === 'undefined') {
+            throw new Error('Missing apostille reference');
+        }
+
+        const {
+            hmacKey,
+            customURLs,
+            casebookCertificate: cert,
+            casebookKey: key,
+        } = req._sails.config;
+        const queryParamsObj = {
+            timestamp: new Date(Date.now()).getTime().toString(),
+            apostilleReference: req.params.apostilleRef,
+        };
+        const queryParams = new URLSearchParams(queryParamsObj);
+        const queryStr = queryParams.toString();
+        const hash = crypto
+            .createHmac('sha512', hmacKey)
+            .update(Buffer.from(queryStr, 'utf-8'))
+            .digest('hex')
+            .toUpperCase();
+
+        return {
+            uri: customURLs.apostilleDownloadAPIURL,
+            agentOptions: {
+                cert,
+                key,
+            },
+            headers: {
+                hash,
+            },
+            qs: queryParamsObj,
+        };
+    },
+
+    _streamFileToClient(options, res) {
+        sails.log.info('Downloading file from Casebook');
+        request
+            .get(options)
```

Comment:
I think you mentioned on engineering it's a deprecated library. Let's avoid if possible. I think it's ok to leave it as it is, unless npm reports a vulnerability related to this.

You may be able to use the native nodejs `http` client instead? 

https://nodejs.org/api/http.html#http_http_request_url_options_callback

Reply:
Good point. I think you've also mentioned this before :)

It's a big change but it makes sense. I'll create a ticket for it.

Reply:
you can always do it bit by bit, or for new features use http.request. Doesn't have to be all in one go, especially if there aren't any critical security issues with `request`